### PR TITLE
Use CodeArtifact for AWS CodeBuild CI jobs

### PR DESCRIPTION
**Pull request by bryceitoc9**
* ~~Doing some additional testing, want to ensure that the buildscripts don't attempt to use CodeArtifact if env vars aren't present (I have confirmed that a CodePipeline tracking the child branch does use CodeArtifact correctly).~~ Confirmed
  * ~~Our main CI and Github Actions/Travis accounts are currently not wired up for CodeArtifact and should skip the CodeArtifact `pre-build` step. Tests run against this PR should confirm this.~~ Confirmed
* ~~We may need to roll back Windows CodeArtifact usage. This is because we can't update our Windows builds to the newer CodeBuild image (required by CodeBuild): https://github.com/microsoft/vscode/issues/77499~~ Rolled back

## Description

This change implements [CodeArtifact](https://aws.amazon.com/codeartifact/) as a cache for NPM for CI jobs run by AWS CodeBuild. This gives us insurance that we can build and deploy VS Code even if NPM goes down.

Additionally, moved to `npm ci` for install to ensure only the packages in the `package-lock` are built.

## Motivation and Context

We don't want to be in a place where we need a fix that we can't deploy due to issues with CI.

## Testing

Tested against a private AWS account. Deploying these changes to a CodePipeline tracking the child branch resulted in a CodeArtifact repo being populated with the appropriate packages.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bryceitoc9** [comment]
Looks like it worked! Executions from my CI look like:
Linux:
```
[Container] 2021/01/06 19:22:22 Entering phase PRE_BUILD
[Container] 2021/01/06 19:22:22 Running command if [ $CODEARTIFACT_DOMAIN ] && [ $CODEARTIFACT_REPO ] && [ $ACCOUNT_ID ]; then
  aws codeartifact login --tool npm --domain "$CODEARTIFACT_DOMAIN" --domain-owner "$ACCOUNT_ID" --repository "$CODEARTIFACT_REPO"
fi

Successfully configured npm to use AWS CodeArtifact repository <URL>
Login expires in 12 hours at 2021-01-07 07:22:27+00:00

[Container] 2021/01/06 19:22:28 Running command npm ci --unsafe-perm
```
Windows:
```
[Container] 2021/01/06 19:24:21 Entering phase PRE_BUILD
[Container] 2021/01/06 19:24:21 Running command if ($Env:CODEARTIFACT_DOMAIN -and $Env:CODEARTIFACT_REPO -and $Env:ACCOUNT_ID) {
  aws codeartifact login --tool npm --domain "$CODEARTIFACT_DOMAIN" --domain-owner "$ACCOUNT_ID" --repository "$CODEARTIFACT_REPO"
}

Successfully configured npm to use AWS CodeArtifact repository <URL>

Login expires in 12 hours at 2021-01-07 07:24:27+00:00

[Container] 2021/01/06 19:24:39 Running command npm ci
```

Logs from these executions skip the success confirmations.

**Review by justinmk3 — APPROVED** [review]

**Comment by bryceitoc9** [comment]
New command correctly hides CodeArtifact URL:
```
2021-01-07T03:56:11.308-08:00	[Container] 2021/01/07 23:56:07 Entering phase PRE_BUILD
2021-01-07T03:56:11.308-08:00	[Container] 2021/01/07 23:56:07 Running command if [ $CODEARTIFACT_DOMAIN ] && [ $CODEARTIFACT_REPO ] && [ $ACCOUNT_ID ]; then
2021-01-07T03:56:11.308-08:00	  aws codeartifact login --tool npm --domain "$CODEARTIFACT_DOMAIN" --domain-owner "$ACCOUNT_ID" --repository "$CODEARTIFACT_REPO" > /dev/null 2>&1
2021-01-07T03:56:11.308-08:00	  if [ $? -eq 0 ]; then
2021-01-07T03:56:11.308-08:00	    echo "Connected to CodeArtifact"
2021-01-07T03:56:11.308-08:00	  else
2021-01-07T03:56:11.308-08:00	    echo "CodeArtifact connection failed. Falling back to npm"
2021-01-07T03:56:11.308-08:00	  fi
2021-01-07T03:56:11.308-08:00	fi
2021-01-07T03:56:11.308-08:00	
2021-01-07T03:56:15.376-08:00	Connected to CodeArtifact
2021-01-07T03:56:15.376-08:00	
2021-01-07T03:56:15.376-08:00	[Container] 2021/01/07 23:56:14 Running command npm ci --unsafe-perm
```

**Review by justinmk3 — APPROVED** [review]

**Review by justinmk3 — APPROVED** [review]

File: buildspec/linuxIntegrationTests.yml [hunk 1]
```diff
@@ -28,6 +28,17 @@ phases:
...
+            # Should only affect tests run through IDEs team-hosted CodeBuild.
+            - |
+                if [ $TOOLKITS_CODEARTIFACT_DOMAIN ] && [ $TOOLKITS_CODEARTIFACT_REPO ] && [ $ACCOUNT_ID ]; then
+                  aws codeartifact login --tool npm --domain "$TOOLKITS_CODEARTIFACT_DOMAIN" --domain-owner "$ACCOUNT_ID" --repository "$TOOLKITS_CODEARTIFACT_REPO" > /dev/null 2>&1
+                  if [ $? -eq 0 ]; then
+                    echo "Connected to CodeArtifact"
+                  else
+                    echo "CodeArtifact connection failed. Falling back to npm"
+                  fi
+                fi
```
Comment (justinmk3) [inline buildspec/linuxIntegrationTests.yml:41]: Please run this block through https://www.shellcheck.net/ . It has several issues. E.g. missing quotes around env vars.

Comment (bryceitoc9) [reply buildspec/linuxIntegrationTests.yml:41]: Got a Powershell equivalent? 😛

Comment (justinmk3) [reply buildspec/linuxIntegrationTests.yml:41]: :) Not sure, but powershell has many fewer gotchas.

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**